src/Uno/Prism.Uno/PrismApplicationBase.Uno.cs

```diff
@@ -0,0 +1,220 @@
+        /// </summary>
+        void InitializeInternal()
+        {
+            ConfigureViewModelLocator();
+            Initialize();
+            OnInitialized();
+        }
+
+        /// <summary>
+        /// Configures the <see cref="Prism.Mvvm.ViewModelLocator"/> used by Prism.
+        /// </summary>
+        protected virtual void ConfigureViewModelLocator()
+        {
+            ViewModelLocationProvider.SetDefaultViewModelFactory((view, type) =>
+            {
+                return Container.Resolve(type);
+            });
+        }
+
+        /// <summary>
+        /// Runs the initialization sequence to configure the Prism application.
+        /// </summary>
+        public virtual void Initialize()
+        {
+            ContainerLocator.SetContainerExtension(CreateContainerExtension);
+            _containerExtension = ContainerLocator.Current;
+            _moduleCatalog = CreateModuleCatalog();
+            RegisterRequiredTypes(_containerExtension);
+            RegisterTypes(_containerExtension);
+            _containerExtension.FinalizeExtension();
+
+            ConfigureModuleCatalog(_moduleCatalog);
+
+            var regionAdapterMappins = _containerExtension.Resolve<RegionAdapterMappings>();
+            ConfigureRegionAdapterMappings(regionAdapterMappins);
+
+            var defaultRegionBehaviors = _containerExtension.Resolve<IRegionBehaviorFactory>();
+            ConfigureDefaultRegionBehaviors(defaultRegionBehaviors);
+
+            RegisterFrameworkExceptionTypes();
+
+            var shell = CreateShell();
+            if (shell != null)
+            {
+                _containerExtension.Resolve<IRegionNavigationService>().NavigationFailed += (s, e) => Console.WriteLine($"Region navigation failed {e.Error}");
+                RegionManager.SetRegionManager(shell, _containerExtension.Resolve<IRegionManager>());
+                RegionManager.UpdateRegions();
+                InitializeShell(shell);
+            }
+
+            InitializeModules();
+        }
+
+        /// <summary>
+        /// Creates the container used by Prism.
+        /// </summary>
+        /// <returns>The container</returns>
+        protected abstract IContainerExtension CreateContainerExtension();
+
+        /// <summary>
+        /// Creates the <see cref="IModuleCatalog"/> used by Prism.
+        /// </summary>
+        ///  <remarks>
+        /// The base implementation returns a new ModuleCatalog.
+        /// </remarks>
+        protected virtual IModuleCatalog CreateModuleCatalog()
+        {
+            return new ModuleCatalog();
+        }
+
+        /// <summary>
+        /// Registers all types that are required by Prism to function with the container.
+        /// </summary>
+        /// <param name="containerRegistry"></param>
+        protected virtual void RegisterRequiredTypes(IContainerRegistry containerRegistry)
+        {
+            containerRegistry.RegisterInstance(_containerExtension);
```

Comment:
this line can be removed. We now register the container when the container initializes itself.